code/game/objects/items/weapons/defib.dm

```diff
@@ -165,8 +165,8 @@
 	if(on)
 		var/M = get(paddles, /mob)
 		remove_paddles(M)
-	. = ..()
 	update_icon()
+	return ..()
```

Comment:
update_icon doesn't appear to do anything special that would warrant calling it before the defib is destroyed. This also doesn't qdel and dereference the paddles and internal battery.